crates/router/src/core/routing.rs

```diff
@@ -2621,3 +2627,67 @@ pub async fn migrate_rules_for_profile(
         errors: error_list,
     })
 }
+
+pub async fn decide_gateway_open_router(
+    state: SessionState,
+    // merchant_context: domain::MerchantContext,
+    req_body: OpenRouterDecideGatewayRequest,
+) -> RouterResponse<DecideGatewayResponse> {
+    let response = SRApiClient::send_decision_engine_request(
+        &state,
+        Method::Post,
+        "decide-gateway",
+        Some(req_body),
+        None,
+        None,
+    )
+    .await
+    .change_context(errors::ApiErrorResponse::InternalServerError)
+    .attach_printable("Failed to perform decide gateway call with open router")?;
+
+    let response_value: serde_json::Value = response
+        .response
+        .get_required_value("response")
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Response from decision engine not obtained")?;
+    let res: DecideGatewayResponse = response_value
+        .parse_value("DecideGatewayResponse")
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed to parse DecidedGateway from response")?;
+
+    Ok(hyperswitch_domain_models::api::ApplicationResponse::Json(
+        res,
+    ))
+}
+
+pub async fn update_gateway_score_open_router(
+    state: SessionState,
+    req_body: UpdateScorePayload,
+) -> RouterResponse<UpdateScoreResponse> {
+    let response = SRApiClient::send_decision_engine_request(
+        &state,
+        Method::Post,
+        "update-gateway-score",
+        Some(req_body),
+        None,
+        None,
+    )
+    .await
+    .change_context(errors::ApiErrorResponse::InternalServerError)
+    .attach_printable("Failed to perform update gateway score call with open router")?;
+
+    let response_value: serde_json::Value = response
+        .response
+        .get_required_value("response")
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Response from decision engine not obtained")?;
+
+    let res: UpdateScoreResponse = response_value
+        .parse_value("UpdateScoreResponse")
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed to parse UpdateScoreResponse from response")?;
```

Comment:
```suggestion
    let response : UpdateScoreResponse = SRApiClient::send_decision_engine_request(
        &state,
        Method::Post,
        "update-gateway-score",
        Some(req_body),
        None,
        None,
    )
    .await?
    .response
    .change_context(errors::ApiErrorResponse::InternalServerError)
    .attach_printable("Failed to perform update gateway score call with open router")?;
```

check if something like this is possible